### PR TITLE
163-edit-activity-status

### PR DESCRIPTION
**Pull request by raduiordache**
@jberends Can you please have a look at it?

### CODE REVIEW

File: pykechain/enums.py [hunk 1]
```diff
@@ -69,3 +69,10 @@ class ComponentXType(Enum):
     SUPERGRID = "superGrid"
     PAGINATEDSUPERGRID = "paginatedSuperGrid"
     FILTEREDGRID = "filteredGrid"
+
+
+class Status(Enum):
```
Comment (jberends) [inline pykechain/enums.py]: lets call this `ActivityStatus` as it bound to actvity. Don;t worry as we will get more statussi.

We might also want to check the 'allowed' transitions in the activity json to ensure that we do not set a status that is invalid to begin with.

File: pykechain/models/activity.py [hunk 2]
```diff
@@ -244,6 +245,13 @@ def edit(self, name=None, description=None, start_date=None, due_date=None, assi
             else:
                 raise TypeError('Assignee should be a string')
 
+        if status:
+            if isinstance(status, (str, text_type)):
+                update_dict.update({'status': status})
+                self.status = status
```
Comment (jberends) [inline pykechain/models/activity.py]: we do not know if the request went through (APIError and stuff) so this is preliminary here

File: pykechain/models/activity.py [hunk 3]
```diff
@@ -244,6 +245,13 @@ def edit(self, name=None, description=None, start_date=None, due_date=None, assi
...
+            else:
+                raise TypeError('Status should be a string')
+
```
Comment (jberends) [inline pykechain/models/activity.py]: We need to add a on L259: 
```python
if status:
    self._json_data['status'] = str(status)

**Comment by raduiordache** [comment]
@jberends I have implemented the feedback, please re-review

**Pull request merged**